packages/ui/src/ReactMarkdown.tsx

```diff
@@ -86,36 +91,42 @@ const MarkdownLink: (navigateNote: NavigateNote) => React.FC<any> =
         );
     };
 
-const flatten = (text, child): any =>
+const flatten = (text: String, child: ElementContent): String =>
     // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
-    typeof child === 'string' ? text + child : React.Children.toArray(child.props.children).reduce(flatten, text);
+    child.type === 'text' ? text + child.value : child.type === 'element' ? child.children.reduce(flatten, text) : '';
 
-const HeadingRenderer: React.FC<any> = ({level, children}) => {
-    const text = children.reduce(flatten, '');
-    const slug = text.toLowerCase().replace(/\W/g, '-');
-    // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
-    return React.createElement('h' + level, {id: slug}, [
-        ...React.Children.toArray(children),
-        <React.Fragment key="link">
-            {level !== 1 ? (
-                <>
-                    &nbsp;
-                    <a className="slug-link" href={`#${slug}`}>
-                        <LinkIcon fontSize="small" />
-                    </a>
-                </>
-            ) : null}
-        </React.Fragment>,
-    ]);
-};
+const getHeadingRenderer =
+    (
+        level: number
+    ): ((
+        props: React.DetailedHTMLProps<React.HTMLAttributes<HTMLHeadingElement>, HTMLHeadingElement> & ExtraProps
+    ) => React.JSX.Element | string | null | undefined) =>
+    ({node, children}) => {
+        const text = node?.children?.reduce(flatten, '');
+        const slug = text?.toLowerCase().replace(/\W/g, '-');
```

Comment:
Warum braucht text ein ?
Wenn es eins braucht, muss dann toLowerCase().replace auch einen nullcheck haben?

Reply:
I set "No headline text" as default value in case the node does not have children. Not sure this will work properly and create a useful changelog entry. But if the node does not have children/text I don't know what would be a good default behavior

Reply:
I am not too familiar with ts. I read` reduce((this), flatten, '')` as: if `!this` then` ''` else `flatten(this)`. If thats the case then default `text` would be empty string. If so then I do not get why ? in ` text?.toLowerCase()`. Ergo reduce works differently and `text` might be null. If thats the case I would assume that `toLowerCase(this)` would not execute if `text` is null as we have `?` to savegueard this in `text?.toLowerCase()`. I would then presume that it passes null onwards, which would collide with `.replace()`
So is there another `?` missing: `text?.toLowerCase()?.replace(/\W/g, '-') `? 
As to the defaults, I presumed from the code so far that `text` may be null - no problem for followup usage. On the other hand, `slug` looks to me as if the default `slug` would be empty string, because if `text` were null `replace` would throw.
But maybe thats all very edge-casy thoughts or even all wrong.... As said earlier: no pro in TS

Reply:
If the node does not exist or the node does not have children, the text will be undefined as reduce will not be called.